internal/appsec/listener/graphqlsec/graphql.go

```diff
@@ -0,0 +1,123 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package graphqlsec
+
+import (
+	"sync"
+	"time"
+
+	"github.com/DataDog/appsec-internal-go/limiter"
+	waf "github.com/DataDog/go-libddwaf/v2"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/appsec/dyngo"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/appsec/emitter/graphqlsec"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/appsec/emitter/sharedsec"
+	listener "gopkg.in/DataDog/dd-trace-go.v1/internal/appsec/listener/sharedsec"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/appsec/trace"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/samplernames"
+)
+
+// GraphQL rule addresses currently supported by the WAF
+const (
+	graphQLServerAllResolversAddr = "graphql.server.all_resolvers"
+	graphQLServerResolverAddr     = "graphql.server.resolver"
+)
```

Comment:
From what I understood from the RFC and discussed with the seceng team: these two addresses are not supposed to be used at the same time per "monitoring listener", but I see them both in the following listener.